src/main/java/org/dice_research/opal/civet/metrics/AvailabilityOfLicensesMetric.java

```diff
@@ -0,0 +1,175 @@
+
+		LOGGER.info("Processing dataset " + datasetUri);
+
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+
+		// NodeIterator nodeIterator = model.listObjectsOfProperty(dataset,
+		// DCAT.keyword);
+
+		// For Score calculation
+		int TotalDistributions = 0;
+		int TotalDistributionsWithLicense = 0;
+		int TotalDistributionsWithRights = 0;
+
+		// score to return
+		int score = 0;
+
+		/*
+		 * For dcat:dataset, we will first check if license is given or not. If not
+		 * check anything with 'right'.
+		 */
+		int NumberOfLicensesInDataset = 0;
+		NodeIterator LicensesIterator = model.listObjectsOfProperty(dataset,DCTerms.license);
+		NodeIterator RightsIterator = model.listObjectsOfProperty(dataset,DCTerms.rights);
+		if (LicensesIterator.hasNext()) {
+           RDFNode licensObject = LicensesIterator.nextNode();
+		   if(!(licensObject.toString().isEmpty()) && isValidLicenseURL(licensObject.toString()))
+			   NumberOfLicensesInDataset++;   
+		}
+		else if (RightsIterator.hasNext()) {
+	           RDFNode RightsObject = RightsIterator.nextNode();
+			   if(!(RightsObject.toString().isEmpty()) && isValidLicenseURL(RightsObject.toString()))
+				   NumberOfLicensesInDataset++;   
+			}
+		
+		
+		if (NumberOfLicensesInDataset == 0) {
+			// This means there is no license/rights info in dct:dataset, we need to check
+			// each dct:distributions
+
+			NodeIterator DistributionsIteratorLicense = model.listObjectsOfProperty(dataset,DCAT.distribution);
+
+			if (DistributionsIteratorLicense.hasNext()) {
+
+				while (DistributionsIteratorLicense.hasNext()) {
+
+					TotalDistributions++;
+					Resource Distribution = (Resource) DistributionsIteratorLicense.nextNode();
+
+					if (Distribution.hasProperty(DCTerms.license)) {
+						// Even if license info is given, check if a valid URI
+
+						if (isValidLicenseURL(Distribution.getProperty(DCTerms.license).getObject().toString()))
+							TotalDistributionsWithLicense++;
+					}
+				}
+			}
+
+			if (TotalDistributionsWithLicense == 0 || TotalDistributionsWithLicense != TotalDistributions)
+
+			{
+
+				/*
+				 * If control came here means, there is no dct:license keyword in available
+				 * distributions or there maybe both dct:license and dct:rights keyword. Check for rights here.
+				 */
+				NodeIterator DistributionsIteratorRights = model.listObjectsOfProperty(dataset,DCAT.distribution);
+
+				while (DistributionsIteratorRights.hasNext()) {
+
+					Resource Distribution = (Resource) DistributionsIteratorRights.nextNode();
+
+					if (Distribution.hasProperty(DCTerms.rights)) {
+						// Even if license info is given, check if a valid URI
+						if (isValidLicenseURL(Distribution.getProperty(DCTerms.rights).getObject().toString()))
+							TotalDistributionsWithRights++;
+					}
+				}
+			}
+		}
+
+		if (NumberOfLicensesInDataset > 0)
+			score = 5;
+		else {
+			if (TotalDistributionsWithLicense == 0 && TotalDistributionsWithRights == 0)
+				score = 0;
```

Comment:
Variable Score is already assigned to 0

Reply:
Any suggestions? Should I just put a single ";" as for doing nothing? I am not sure :D

Reply:
no need to cover those cases where the score is getting assigned to 0 again because it is already 0

Reply:
Okay, I removed those cases.